tests/test_all_configurations.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+#
+# Try all test configurations, typically before a release. This script should be
+# executed from the root of a fresh local clone of the libavif repository, for
+# example (Unix only):
+#   git clone https://github.com/AOMediaCodec/libavif.git
+#   cd libavif
+#   tests/test_all_configurations.sh
+
+set -e
+
+pushd ext
+  rm -rf googletest
+  ./googletest.cmd
+popd
+
+rm -rf build_*
+
+for BUILD_TYPE in Debug Release; do
+  pushd ext
+    rm -rf aom
+    ./aom.cmd
+    rm -rf dav1d
+    ./dav1d.cmd
+    rm -rf libyuv
+    ./libyuv.cmd
+    rm -rf libwebp
+    ./libsharpyuv.cmd
+  popd
+
+  mkdir build_${BUILD_TYPE}
+  pushd build_${BUILD_TYPE}
+    cmake .. \
+     -DAVIF_BUILD_APPS=ON -DAVIF_BUILD_EXAMPLES=ON \
+     -DAVIF_BUILD_TESTS=ON -DAVIF_ENABLE_GTEST=ON -DAVIF_LOCAL_GTEST=ON \
+     -DAVIF_CODEC_AOM=ON -DAVIF_LOCAL_AOM=ON \
+     -DAVIF_CODEC_DAV1D=ON -DAVIF_LOCAL_DAV1D=ON \
+     -DAVIF_LOCAL_LIBYUV=ON -DAVIF_LOCAL_LIBSHARPYUV=ON \
+     -DCMAKE_BUILD_TYPE=${BUILD_TYPE} -DBUILD_SHARED_LIBS=OFF \
+     -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++
+    cmake --build . --parallel
+    ctest . -j $(nproc)
+  popd
+
+  # The memory sanitizer triggers too many use-of-uninitialized-value errors
+  # (possibly false positives because standard libraries were not instrumented).
+  for SANITIZER in address thread undefined; do
+    pushd ext
+      rm -rf aom
+      CC=clang CXX=clang++ CFLAGS=-fsanitize=${SANITIZER} CXXFLAGS=-fsanitize=${SANITIZER} LDFLAGS=-fsanitize=${SANITIZER} \
+      ./aom.cmd
```

Comment:
For libaom, the best way is to patch aom.cmd so that we pass `-DSANITIZE=${SANITIZER}` to the cmake command. This should replace the `CFLAGS`, `CXXFLAGS`, and `LDFLAGS` overrides.

Reply:
Why is it "the best way"?

Reply:
It is the best way in the sense that it is how the libaom team builds libaom for sanitizers:
https://aomedia.googlesource.com/aom/+/master/README.md#sanitizers

So it is guaranteed to work. Perhaps it would be better to call it the "official way".

Reply:
It does not seem to be done in https://github.com/AOMediaCodec/libavif/blob/main/tests/oss-fuzz/build.sh so leaving it as is for now.